packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.controller.js

```diff
@@ -8,48 +8,72 @@ export default class {
   }
 
   $onInit() {
+    this.trackDashboard('users::add_a_user', 'page');
     this.availableRoles = [...this.roles];
     this.model = {
       username: '',
       password: '',
+      categories: [],
+      channels: [],
+      commands: [],
+      keys: [],
       selectedRoles: [],
     };
+    this.isRolesReadOnly = !this.isFeatureActivated('getRoles');
+    if (this.isRolesReadOnly) {
+      this.model.selectedRoles = [];
+    }
+  }
+
+  getUserFromModel() {
+    const user = {
+      name: this.model.username,
+    };
+    if (this.database.engine === 'redis') {
```

Comment:
Move the engine types to a constant file

Reply:
Done here: https://github.com/ovh/manager/pull/5488/commits/45d16da89ff2c70503a9a111c3fd5996a5eecbe1